transport/serialize/serializer.go

```diff
@@ -32,6 +31,8 @@ type Serialization int
 type Serializer interface {
 	Serialize(wamp.Message) ([]byte, error)
 	Deserialize([]byte) (wamp.Message, error)
+	SerializeDataItem(item interface{}) ([]byte, error)
+	DeserializeDataItem([]byte, interface{}) (interface{}, error)
```

Comment:
I'm not sure the generic returned interface is useful in any way. Why not just do this:

```suggestion
	DeserializeDataItem([]byte, interface{}) error
```

Then if someone wants an inteface{} they can always call:
```
var generic interface{}
if err := s.DeserializeDataItem(data, &generic); err != ni{
    // foo
}
```